simplivity/resources/datastores.py

```diff
@@ -113,3 +113,8 @@ def __init__(self, connection, resource_client, data):
         self.data = data
         self._connection = connection
         self._client = resource_client
+
+    def delete(self, timeout=-1):
+        """Deletes a datastore."""
+        resource_uri = "{}/{}".format(URL, self.data["id"])
+        self.data = self._client.do_delete(resource_uri, timeout, None)
```

Comment:
I think we should not overwrite resource data here?
My suggestion is to keep an if condition here to check the Datastore id in the affected resource list(output of do_delete).
If Datastore id presents in the affected resource list, make self.data = None and return True. 
If Datastore id doesn't exist in the affected resource list, the delete operation is failed and we don't have to overwrite the resource data, return False

Reply:
I agree that I should not be overwriting the self.data

The datastore delete operation returns a task response; however, the "affected_objects": field is empty within the task object
```
DEBUG:Task details: {'state': 'IN_PROGRESS', 'id': '4229bfea-777c-6a1e-f72c-c94e80d4e2e0:4229bfea-777c-6a1e-f72c-c94e80d4e2e0:859fb642-16aa-42c4-b19b-f32b0175de6b', 'name': '', 'type': 'com.simplivity.task.vmware.datastore.delete.controller', 'owner_id': '4229bfea-777c-6a1e-f72c-c94e80d4e2e0', 'destination_id': '4229bfea-777c-6a1e-f72c-c94e80d4e2e0', 'start_time': '2020-01-23T15:21:36Z', 'end_time': 'NA', 'percent_complete': 25, 'error_code': 0, 'affected_objects': []}
```
```
DEBUG:Task details: {'state': 'COMPLETED', 'id': '4229bfea-777c-6a1e-f72c-c94e80d4e2e0:4229bfea-777c-6a1e-f72c-c94e80d4e2e0:859fb642-16aa-42c4-b19b-f32b0175de6b', 'name': '', 'type': 'com.simplivity.task.vmware.datastore.delete.controller', 'owner_id': '4229bfea-777c-6a1e-f72c-c94e80d4e2e0', 'destination_id': '4229bfea-777c-6a1e-f72c-c94e80d4e2e0', 'start_time': '2020-01-23T15:21:36Z', 'end_time': '2020-01-23T15:21:37Z', 'percent_complete': 100, 'error_code': 0, 'affected_objects': []}
```

It appears that the task object will handle the following exceptions
- task timeout value has been reached
- task failed to complete due to an error, i.e. "You cannot delete a datastore that is used by VMs. Migrate or remove VMs and try again."

How would you prefer to proceed?

Reply:
You are correct, the task module is handling most of the request errors.
As delete operation doesn't return the resource id, we can just set self.data as None once the delete operation is successful.